ta4j-core/src/main/java/org/ta4j/core/analysis/Returns.java

```diff
@@ -0,0 +1,176 @@
+/*******************************************************************************
+ *   The MIT License (MIT)
+ *
+ *   Copyright (c) 2014-2017 Marc de Verdelhan, 2017-2018 Ta4j Organization 
+ *   & respective authors (see AUTHORS)
+ *
+ *   Permission is hereby granted, free of charge, to any person obtaining a copy of
+ *   this software and associated documentation files (the "Software"), to deal in
+ *   the Software without restriction, including without limitation the rights to
+ *   use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ *   the Software, and to permit persons to whom the Software is furnished to do so,
+ *   subject to the following conditions:
+ *
+ *   The above copyright notice and this permission notice shall be included in all
+ *   copies or substantial portions of the Software.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ *   FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ *   COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ *   IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *******************************************************************************/
+package org.ta4j.core.analysis;
+
+import org.ta4j.core.Indicator;
+import org.ta4j.core.TimeSeries;
+import org.ta4j.core.Trade;
+import org.ta4j.core.TradingRecord;
+import org.ta4j.core.num.DoubleNum;
+import org.ta4j.core.num.NaN;
+import org.ta4j.core.num.Num;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * The return rates.
+ * </p>
+ * This class allows to compute the return rate of a price time-series
+ */
+public class Returns implements Indicator<Num> {
+
+    public enum ReturnType {
+        LOG {
+            public Num calculate(Num x_new, Num x_old) {
+                // r_i = ln(P_i/P_(i-1))
+                return ((DoubleNum) x_new.dividedBy(x_old)).log();
+            }
+        },
+        ARITHMETIC {
+            @Override
+            public Num calculate(Num x_new, Num x_old) {
+                // r_i = P_i/P_(i-1) - 1
+                return x_new.dividedBy(x_old).minus(x_new.numOf(1));
```

Comment:
Performance would be better if you store `numOf(1)` as a class field instead of converting `int` to `num ` in every calculation